test/cppsuite/tests/cursor_bound_01.cpp

```diff
@@ -0,0 +1,586 @@
+    {
+        std::string random_key;
+        uint32_t rollback_retries = 0;
+
+        while (tc->running()) {
+
+            collection &coll = tc->db.get_random_collection();
+            scoped_cursor cursor = tc->session.open_scoped_cursor(coll.name);
+            tc->transaction.begin();
+
+            while (tc->transaction.active() && tc->running()) {
+
+                /* Generate a random key. */
+                random_key = random_generator::instance().generate_random_string(tc->key_size);
+
+                /* Call search near to position cursor. */
+                int exact;
+                cursor->set_key(cursor.get(), random_key);
+                int ret = cursor->search_near(cursor.get(), &exact);
+                if (ret == WT_NOTFOUND)
+                    continue;
+
+                /* Retrieve the key the cursor is pointing at. */
+                const char *key;
+                testutil_check(cursor->get_key(cursor.get(), &key));
+
+                /* Perform the operation on the key. */
+                if ((tc->*op_func)(cursor, coll.id, key)) {
+                    if (tc->transaction.can_commit()) {
+                        /* We are not checking the result of commit as it is not necessary. */
+                        if (tc->transaction.commit())
+                            rollback_retries = 0;
+                        else
+                            ++rollback_retries;
+                    }
+                } else {
+                    tc->transaction.rollback();
+                    ++rollback_retries;
+                }
+                testutil_assert(rollback_retries < MAX_ROLLBACKS);
+
+                /* Sleep the duration defined by the configuration. */
+                tc->sleep();
+            }
+
+            /* Rollback any transaction that could not commit before the end of the test. */
+            if (tc->transaction.active())
+                tc->transaction.rollback();
+
+            /* Reset our cursor to avoid pinning content. */
+            testutil_check(cursor->reset(cursor.get()));
+        }
+    }
+
+    /* 
+     * Use the random generator either set no bounds, only lower bounds, only upper bounds or both
+     * bounds on the range cursor. The lower and upper bounds are randomly generated strings as well.
+     * The inclusive configuration is also randomly set as well.
+     */
+    std::pair<std::string, std::string> set_random_bounds(test_harness::thread_context *tc, scoped_cursor &range_cursor) {
+        int set_random_bounds;
+        int64_t key_size;
+        std::string lower_key, upper_key;
+
+
+        set_random_bounds = random_generator::instance().generate_integer(0, 3);
+        if (set_random_bounds == LOWER_BOUND_SET || set_random_bounds == ALL_BOUNDS_SET) {
+            key_size = random_generator::instance().generate_integer(
```

Comment:
Seems like you don't need this, could be a work in progress? Same for the next few lines.